homeassistant/components/mqtt/__init__.py

```diff
@@ -245,42 +195,28 @@ async def _async_config_entry_updated(hass: HomeAssistant, entry: ConfigEntry) -
     await hass.config_entries.async_reload(entry.entry_id)
 
 
-async def async_fetch_config(
-    hass: HomeAssistant, entry: ConfigEntry
-) -> dict[str, Any] | None:
+async def async_fetch_config(hass: HomeAssistant, entry: ConfigEntry) -> dict[str, Any]:
     """Fetch fresh MQTT yaml config from the hass config."""
     mqtt_data = get_mqtt_data(hass)
     hass_config = await conf_util.async_hass_config_yaml(hass)
     mqtt_data.config = PLATFORM_CONFIG_SCHEMA_BASE(hass_config.get(DOMAIN, {}))
+    try:
+        entry_config = CONFIG_SCHEMA_ENTRY(dict(entry.data))
+    except vol.MultipleInvalid as ex:
+        raise ConfigEntryError(
+            f"The MQTT config entry is invalid, please correct it: {ex}"
+        ) from ex
```

Comment:
When can this happen? How can the user correct it?

Reply:
This replaces the function where the broker setting or other invalid setting could be in the entry.
The user can correct it using the UI.

Reply:
If e.g. the broker setting is missing (e.g. when some old import was used) and the schema is not valid it will throw.
The user sees this (left broker setting await):

![afbeelding](https://user-images.githubusercontent.com/7188918/229425416-86cf039e-12c5-48af-a799-148c2041b674.png)

From this point the user can correct the entry via the UI.
Further the entry is no longer automatically updated anymore, but only via the UI.

Reply:
OK 👍